specs/logging_implementation.adoc

```diff
@@ -0,0 +1,202 @@
+     message_uuid (for log flight operation job id is uuid, for alert alert_id is uuid
+                   for normal log messages this field is optional)
+     severity(optional for log and flight operation)
+     tag(optional for log and flight operation)
+
+     timestamp(generated and appended by handler automatically)
+    (Before log any message filename, function_name, line_number appended with message
+    by handler).
+
+    1. For log - message_handler(log_handler_name, message_type, log_level, message,
+       message_trace_back_flag)
+    2. For flight operation - message_handler(log_handler_name, message_type, log_level, message,
+       message_trace_back_flag, message_uuid(job_id))
+    3. For alert - message_handler(log_handler_name, message_type, log_level, message,
+       message_trace_back_flag, message-uuid(alert_id), severity, tag)
+```
+* Create a new function called message_handler in Message class and process the
+  object values.
+* If the message type is Alert then store the message in log file with uuid (log level
+  for alert is always warning). and put the object in node_agent socket.
+* If the message type is flight operation then log the message as INFO with message
+  uuid (uuid is used to trace the filght operation).
+* If the type is Log then store the message in log file based on log level.
+
+=== Tendrl/node_agent impact
+
+* Create socket to receive alert.
+* Import message class from common and change all log statement (it will be in another PR).
+
+=== Tendrl/gluster_integration impact
+
+* Import message class from common and change all log statement (it will be in another PR).
+
+=== Tendrl/ceph_integration impact
+
+* Import message class from common and change all log statement (it will be in another PR).
+
+=== Tendrl/alerting impact
+
+* Import message class from common and change all log statement (it will be inanother PR).
+
+=== Tendrl/performance_monitoring impact
+
+* Import message class from common and change all log statement (it will be in another PR).
+
+== Security impact
+
+None
+
+== Notifications/Monitoring impact
+
+None
+
+== Other end user impact
+
+None
+
+== Performance Impact
+
+None
+
+== Other deployer impact
+
+None
+
+== Developer impact
+
+None
+
+== Implementation
+
+* Create a new class called "Message" in common for message handling with attributes
+  handler_name, message_type, message_uuid, log_level, message_tag, message, severity,
+  timestamp. (other attributes are filename, function_name, line_number. These are
+  populated by message handler automatically).
+  (Logging will take line number and function name from where it is called, Here
+  the problem is we are logging the message from common so it takes a function name
+  and line number from common only. So it makes debugging complex. To avoid this
+  we have to find the function name, line number, file name and append this data
+  with message manually).
+  ```
+  from inspect import getframeinfo, stack
```

Comment:
This is not looking properly in html generated from adoc use asciidoctor <file_name>.adoc after installing asciidoctor. The command generates a html file in current directory and that you can open in any browser and check it. You probably need to leave a line before ```` or not sure if ```` is a valid symbol in adoc. Please check it.